### ISSUE TITLE
PIPNet training and noise

### ISSUE THREAD

**Issue by nikopj** [sharif-apu/BJDD_CVPR21#3]
In dataTools/customTransform.py, line 13, you have the line
```
noisyTensor = tensor + torch.randn(tensor.size()).uniform_(0, 1.) * sigma  + self.mean
```
This looks to be adding *uniform* random noise (U(0,sigma)) instead of *Gaussian* random noise (N(0,sigma^2)), as described in the paper ([see PyTorch docs](https://pytorch.org/docs/1.5.0/tensors.html?highlight=uniform_#torch.Tensor.uniform_)). This function is used in dataTools/customDataloader.py (customDatasetReader) which is then used in the main training code (mainModule/BJDD.py, lines 124, 91). Are the numbers reported in the paper derived from this code?

Another question: are the numbers reported in the paper a single model PIPNet model trained over a range of noise-levels, or multiple models trained at each level? If it's a single model, what was the training noise-level range?

Thanks,
Nikola

**Comment by sharif-apu** [comment]
Hi Nikola!
Due to implementation constraints, we used an uniform distribution right after the Gaussian distribution. We did it purposefully, thus the sigma can be controlled on the normalized data. However, you may modify that chunk of code as per your requirements. And yes, we used this code to produce all reported results with the same principle.

We used a range of noise (as described in the paper) to train a single model and produced the results accordingly. Please see line 30 of dataTools/customDataloader.py.
